### PR TITLE
PLPO-30: Implement backend services for login purposes.

### PR DESCRIPTION
**Pull request by dczyz**

### CODE REVIEW

File: planning-poker-app/src/main/resources/application.properties [hunk 1]
```diff
@@ -0,0 +1,2 @@
+jira.uri=https://futureprocessing.jira.com
```
Comment (lkulig) [inline planning-poker-app/src/main/resources/application.properties]: why????

File: planning-poker-app/src/main/java/com/futureprocessing/planningpoker/controller/URLMappingController.java [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+import static com.futureprocessing.planningpoker.security.UserRole.CROUPIER;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+
+@Controller
+@RequestMapping("/")
+public class URLMappingController {
```
Comment (lkulig) [inline planning-poker-app/src/main/java/com/futureprocessing/planningpoker/controller/URLMappingController.java]: URL Mapping, maybe some other naming?

File: planning-poker-app/src/main/resources/application.properties (hunk 1 above)
Comment (activey) [inline planning-poker-app/src/main/resources/application.properties]: why not? how would you pass the jira url?

File: planning-poker-app/src/main/java/com/futureprocessing/planningpoker/security/AuthenticationProviderImpl.java [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+import org.springframework.security.authentication.AuthenticationCredentialsNotFoundException;
+import org.springframework.security.authentication.AuthenticationProvider;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.AuthenticationException;
+import org.springframework.security.core.authority.SimpleGrantedAuthority;
+import org.springframework.stereotype.Component;
+
+@Component
+public class AuthenticationProviderImpl implements AuthenticationProvider {
```
Comment (activey) [inline planning-poker-app/src/main/java/com/futureprocessing/planningpoker/security/AuthenticationProviderImpl.java]: I would suggest a different naming, like PlanningPokerAP? ApplicationAP? this impl doesnt look nice imho

File: planning-poker-app/src/main/java/com/futureprocessing/planningpoker/service/ContentService.java [hunk 4]
```diff
@@ -0,0 +1,6 @@
+package com.futureprocessing.planningpoker.service;
+
+public interface ContentService {
```
Comment (activey) [inline planning-poker-app/src/main/java/com/futureprocessing/planningpoker/service/ContentService.java]: hm, how about different interface name? AuthenticationService? ContentService could be just limited to operations related to ... content ;)

File: planning-poker-app/src/main/java/com/futureprocessing/planningpoker/service/JiraContentService.java [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+import org.springframework.context.annotation.Scope;
+import org.springframework.stereotype.Component;
+import net.rcarz.jiraclient.BasicCredentials;
+import net.rcarz.jiraclient.JiraClient;
+import net.rcarz.jiraclient.JiraException;
+import java.util.Locale;
+
+@Component
+@Scope(value = "session", proxyMode = TARGET_CLASS)
+public class JiraContentService implements ContentService {
```
Comment (activey) [inline planning-poker-app/src/main/java/com/futureprocessing/planningpoker/service/JiraContentService.java]: how about moving it to a separate gradle module?

File: planning-poker-app/src/test/java/com/futureprocessing/planningpoker/service/JiraContentServiceImplTest.java [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+import com.futureprocessing.planningpoker.AbstractUnitTest;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.Spy;
+import net.rcarz.jiraclient.JiraClient;
+import net.rcarz.jiraclient.JiraException;
+
+public class JiraContentServiceImplTest extends AbstractUnitTest {
```
Comment (activey) [inline planning-poker-app/src/test/java/com/futureprocessing/planningpoker/service/JiraContentServiceImplTest.java]: naming change will affect this little guy here as well

File: planning-poker-app/src/main/java/com/futureprocessing/planningpoker/Application.java [hunk 7]
```diff
@@ -1,22 +1,38 @@
...
+import org.springframework.boot.builder.SpringApplicationBuilder;
+import org.springframework.boot.context.web.SpringBootServletInitializer;
 import org.springframework.context.ApplicationContext;
-import java.util.Arrays;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.ImportResource;
 
 @SpringBootApplication
-public class Application {
+@ImportResource("classpath*:applicationContext.xml")
```
Comment (activey) [inline planning-poker-app/src/main/java/com/futureprocessing/planningpoker/Application.java]: do we need this xml?

File: planning-poker-jira-backend-api/src/test/java/com/futureprocessing/planningpoker/jira/AbstractUnitTest.java [hunk 8]
```diff
@@ -0,0 +1,18 @@
+package com.futureprocessing.planningpoker.jira;
+
+import static org.mockito.MockitoAnnotations.initMocks;
+
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.rules.ExpectedException;
+
+public class AbstractUnitTest {
```
Comment (activey) [inline planning-poker-jira-backend-api/src/test/java/com/futureprocessing/planningpoker/jira/AbstractUnitTest.java]: I believe we will use it in other modules as well, extrac to 'commons'? ;]

File: planning-poker-app/build.gradle [hunk 9]
```diff
@@ -11,13 +11,20 @@ buildscript {
...
+configurations {
+    providedRuntime
+}
 
 dependencies {
-    compile('org.springframework.boot:spring-boot-starter-web') {
-        exclude module: 'spring-boot-starter-tomcat'
-    }
-    compile('org.springframework.boot:spring-boot-starter-jetty')
+    compile project(':planning-poker-backend-api')
```
Comment (activey) [inline planning-poker-app/build.gradle]: is there anything in that module? cant find any files

File: planning-poker-jira-backend-api/src/test/java/com/futureprocessing/planningpoker/jira/AbstractUnitTest.java (hunk 8 above)
Comment (dczyz) [inline planning-poker-jira-backend-api/src/test/java/com/futureprocessing/planningpoker/jira/AbstractUnitTest.java]: Ok.

File: planning-poker-app/src/main/java/com/futureprocessing/planningpoker/Application.java [hunk 10]
```diff
@@ -1,22 +1,38 @@
...
+    @Override
+    protected SpringApplicationBuilder configure(SpringApplicationBuilder application) {
+        return application.sources(Application.class);
+    }
 
     public static void main(String[] args) {
         ApplicationContext ctx = SpringApplication.run(Application.class, args);
 
-        System.out.println("Let's inspect the beans provided by Spring Boot:");
+        LOGGER.info("Let's inspect the beans provided by Spring Boot:");
```
Comment (pbenecki) [inline planning-poker-app/src/main/java/com/futureprocessing/planningpoker/Application.java]: Actually we could get rid of it, spring health services do something similar. It was only copied from an example from spring boot docs to not have an empty project.

Comment (dczyz) [inline planning-poker-app/src/main/java/com/futureprocessing/planningpoker/Application.java]: Ok.

**Pull request merged**